src/routers/semester.ts

```diff
@@ -1,24 +1,53 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/semesterOperations';
 
 const semesterRouter = express.Router();
 
-semesterRouter.get('/', (req, res) => {
-	console.log('Get Semester');
-	res.send('Get semesterRouter');
+semesterRouter.get('/', async (req, res) => {
+	try {
+		console.log('Get Semester');
+		const semester = await read(req.body.id);
+		const resData = {
+			...semester,
+			courses: Array.from(semester?.courses || []),
+		};
+		res.send(resData);
+	} catch (err) {
+		console.log(err);
+		res.send(err);
```

Comment:
On the res.send(err) we want to make sure that we also change the error code I think.. I am not sure which code is best for this kind of failure though ahha

Reply:
I need to make some changes to how I sent all errors - I'm sending custom strings instead of the actual errors!

Reply:
Actually, it is best practice most of the time to send a generic error rather than the specifics of what occurred. It can just be `new Error("Failed to get course with ID ---x---")` we **do not want to** differentiate between the user not being allowed to see the course, or the course not existing. The purpose is security. 

Check this article out if you want to learn more about that! https://cheatsheetseries.owasp.org/cheatsheets/Error_Handling_Cheat_Sheet.html

Reply:
Should the errors include codes like 400 / 403 / etc...?

Reply:
**We should use 404 every time** (not found).

 We **do not want to send** people 403's ever since we don't want to tell people that the object exists and they just aren't allowed to get/edit/delete it. This is following good security practices.